sdk/clientcore/core/src/main/java/io/clientcore/core/utils/IOExceptionCheckedConsumer.java

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package io.clientcore.core.utils;
+
+import java.io.IOException;
+
+/**
+ * A consumer checked with {@link IOException}.
+ *
+ * @param <T> The type of the consumer.
+ */
+@FunctionalInterface
+public interface IOExceptionCheckedConsumer<T> {
```

Comment:
does it need to be public? 
Maybe we can have a `CoreException` contract on a lower level so that we don't need to expose those?

Reply:
We could, though this is also very convenient for developing in a lot of places where we want a callback like `bytes -> stream.write(bytes)` where we don't need the caller to try / catch.

Reply:
let's keep it then